data/caasp/update.sh

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env bash
+set -ex
+
+# 2.0 -> 2.0 http://download.suse.de/ibs/Devel:/CASP:/2.0:/ControllerNode:/TestUpdate/standard/Devel:CASP:2.0:ControllerNode:TestUpdate.repo
+# 
+
+while getopts "r:" opt; do
+case $opt in
+	r)
+		REPO=$OPTARG
+		;;
+	:)
+		echo "Option -$opt requires an argument." >&2
+		;;
+esac
+done
+
+saltid=$(docker ps | grep salt-master | awk '{print $1}')
+where="-P roles:(admin|kube-(master|minion))"
+srun="docker exec -i $saltid salt --batch 7"
+runner="$srun $where cmd.run"
+
+# Remove non-existent ISO repository
+# $runner "zypper rr 1"
+
+# Add repository as system venodors
+$runner 'echo -e "[main]\nvendors = suse,opensuse,obs://build.suse.de,obs://build.opensuse.org" > /etc/zypp/vendors.d/vendors.conf'
+$runner "zypper ar --refresh --no-gpgcheck $REPO"
+
+# Manually Trigger Transactional Update (or wait up to 24 hours for it run by itself)
+$runner 'systemctl disable --now transactional-update.timer'
+$runner '/usr/sbin/transactional-update cleanup dup salt'
+
+# Manually Refresh the Grains (or wait up to 10 minutes)
+$srun '*' saltutil.refresh_grains
+
+
+
+# Check after update (reboot)
+
+# # caasp-container-manifests
+# # check installed version
+# zypper if --provides caasp-container-manifests | grep 100.0.0
+# # check there is a "fo:bar" in the public manifests
+# grep "fo: bar" /usr/share/caasp-container-manifests/public.yaml
+# # check the change has been "activated"
+# grep "fo: bar" /etc/kubernetes/manifests/public.yaml
+# 
+# # container-feeder
+# # check installed version
+# zypper if --provides container-feeder | grep 100.0.0
+# # check changes in README - zypper config changed to exclude doc
+# # grep "This image has been updated fine" /usr/share/doc/packages/container-feeder/README.md
+# 
+# # kubernetes-salt
+# # check version
+# zypper if --provides kubernetes-salt | grep 100.0
+# # check etc/motd in a minion node
+# COMMAND="grep 'This has been updated fine' /etc/motd"
+# docker exec -i $saltid salt --batch 20 -P "roles:kube-minion" cmd.run "$COMMAND" | grep 'This has been updated fine'
+# 
+# 
+# # velum image
```

Comment:
There's a [bug](https://bugzilla.suse.com/show_bug.cgi?id=1066655) that needs to be fixed about it. However, we have to find out how to verify this ourselves. I guess there's must be a value in *salt* that holds the result of the *update per node*.

Reply:
With "docker exec -i $container_id ls /IMAGE_UPDATED" I check that we are running updated velum - as you say, there is no GUI notification yet